### PR TITLE
patching for 4.2

### PR DESCRIPTION
**Pull request by hubrix**

### CODE REVIEW

**Comment by hubrix** [comment]
@greyblake this appears to now work correctly in 4.2 at least with our specific usage

File: lib/pg_saurus/create_index_concurrently.rb [hunk 1]
```diff
@@ -70,7 +70,6 @@ module Migration
     #
     # @see ActiveRecord::ConnectionAdapters::SchemaStatements.add_index in pg_saurus gem
     def add_index(table_name, column_name, options = {}, &block)
-      table_name = ::ActiveRecord::Migrator.proper_table_name(table_name)
```
Comment (albertosaurus) [inline lib/pg_saurus/create_index_concurrently.rb]: Does this break Rails 4.1?

Comment (hubrix) [inline lib/pg_saurus/create_index_concurrently.rb]: its unclear, i am not sure why it needed this in the first place

**Comment by hubrix** [comment]
we are no able to run on 4.2 with this except add_foreign_key migrations fail

**Comment by albertosaurus** [comment]
Rails 4.2 introduced its own foreign key support which conflicts with this. I haven't yet had time to investigate what would be involved in implementing a fix.

**Comment by hubrix** [comment]
could we simply remove it from pg_saurus?
ᐧ

**Comment by albertosaurus** [comment]
Possibly. We may wind up doing PgSaurus 2.X which will require Rails 4.2+.

**Comment by albertosaurus** [comment]
How soon do you folks need 4.2 support?

**Comment by hubrix** [comment]
we're on 4.2 now and other than new foreign key migrations thingss seem ok,
i might try to patch that bit when i really need it. we had to move to 4.2
because there is a bug with pg gem that doesnt allows us to save certain
files to bytea and it's fixed in 4.2/pg.18
ᐧ

**Comment by baburdick** [comment]
I get this failure when running specs under Rails 4.1 with these changes merged in:
```
  == 20120904114121 AddConcurrentlyIndexWithForeignKey: migrating ===============
  rake aborted!
  StandardError: An error has occurred, this and all later migrations canceled:
  
  undefined method `gsub' for :pets:Symbol~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/connection_adapters/postgresql_adapter/foreigner_methods.rb:200:in `foreign_key_name'
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/connection_adapters/postgresql_adapter/foreigner_methods.rb:124:in `add_foreign_key_sql'
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/connection_adapters/postgresql_adapter/foreigner_methods.rb:100:in `add_foreign_key'
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/create_index_concurrently.rb:129:in `add_foreign_key'
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/spec/dummy/db/migrate/20120904114121_add_concurrently_index_with_foreign_key.rb:3:in `change'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:602:in `exec_migration'
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/migration/set_role_method.rb:71:in `exec_migration_with_role'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:586:in `block (2 levels) in migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:585:in `block in migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/connection_adapters/abstract/connection_pool.rb:294:in `with_connection'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:584:in `migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:759:in `migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:998:in `block in execute_migration_in_transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:1044:in `block in ddl_transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/connection_adapters/abstract/database_statements.rb:201:in `block in transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/connection_adapters/abstract/database_statements.rb:209:in `within_new_transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/connection_adapters/abstract/database_statements.rb:201:in `transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/transactions.rb:208:in `transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:1044:in `ddl_transaction'
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/create_index_concurrently.rb:205:in `ddl_transaction_with_postponed_queries'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:997:in `execute_migration_in_transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:959:in `block in migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:955:in `each'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:955:in `migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:814:in `up'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:792:in `migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/railties/databases.rake:34:in `block (2 levels) in <top (required)>'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/bin/ruby_executable_hooks:15:in `eval'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/bin/ruby_executable_hooks:15:in `<main>'
  NoMethodError: undefined method `gsub' for :pets:Symbol
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/connection_adapters/postgresql_adapter/foreigner_methods.rb:200:in `foreign_key_name'
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/connection_adapters/postgresql_adapter/foreigner_methods.rb:124:in `add_foreign_key_sql'
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/connection_adapters/postgresql_adapter/foreigner_methods.rb:100:in `add_foreign_key'
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/create_index_concurrently.rb:129:in `add_foreign_key'
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/spec/dummy/db/migrate/20120904114121_add_concurrently_index_with_foreign_key.rb:3:in `change'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:602:in `exec_migration'
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/migration/set_role_method.rb:71:in `exec_migration_with_role'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:586:in `block (2 levels) in migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:585:in `block in migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/connection_adapters/abstract/connection_pool.rb:294:in `with_connection'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:584:in `migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:759:in `migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:998:in `block in execute_migration_in_transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:1044:in `block in ddl_transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/connection_adapters/abstract/database_statements.rb:201:in `block in transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/connection_adapters/abstract/database_statements.rb:209:in `within_new_transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/connection_adapters/abstract/database_statements.rb:201:in `transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/transactions.rb:208:in `transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:1044:in `ddl_transaction'
  ~/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/create_index_concurrently.rb:205:in `ddl_transaction_with_postponed_queries'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:997:in `execute_migration_in_transaction'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:959:in `block in migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:955:in `each'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:955:in `migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:814:in `up'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/migration.rb:792:in `migrate'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.1.9/lib/active_record/railties/databases.rake:34:in `block (2 levels) in <top (required)>'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/bin/ruby_executable_hooks:15:in `eval'
  ~/.rvm/gems/ruby-2.1.5@pg_saurus/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by baburdick** [comment]
I get this failure when running specs under Rails 4.2 with these changes merged in:
```
== 20120207150844 AddForeignKeys: migrating ===================================
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

undefined method `proper_table_name' for ActiveRecord::Migrator:Class/Users/bruceb/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/connection_adapters/postgresql_adapter/foreigner_methods.rb:131:in `add_foreign_key_sql'
/Users/bruceb/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/connection_adapters/postgresql_adapter/foreigner_methods.rb:100:in `add_foreign_key'
/Users/bruceb/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/create_index_concurrently.rb:129:in `add_foreign_key'
/Users/bruceb/Code/OSS/git-repos/HornsAndHooves-pg_saurus/spec/dummy/db/migrate/20120207150844_add_foreign_keys.rb:4:in `change'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:606:in `exec_migration'
/Users/bruceb/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/migration/set_role_method.rb:71:in `exec_migration_with_role'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:590:in `block (2 levels) in migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:589:in `block in migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:588:in `migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:765:in `migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:995:in `block in execute_migration_in_transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:1041:in `block in ddl_transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `block in transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/connection_adapters/abstract/transaction.rb:188:in `within_new_transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/transactions.rb:220:in `transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:1041:in `ddl_transaction'
/Users/bruceb/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/create_index_concurrently.rb:205:in `ddl_transaction_with_postponed_queries'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:994:in `execute_migration_in_transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:956:in `block in migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:952:in `each'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:952:in `migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:820:in `up'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:798:in `migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/tasks/database_tasks.rb:137:in `migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/railties/databases.rake:44:in `block (2 levels) in <top (required)>'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/bin/ruby_executable_hooks:15:in `eval'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/bin/ruby_executable_hooks:15:in `<main>'
NoMethodError: undefined method `proper_table_name' for ActiveRecord::Migrator:Class
/Users/bruceb/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/connection_adapters/postgresql_adapter/foreigner_methods.rb:131:in `add_foreign_key_sql'
/Users/bruceb/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/connection_adapters/postgresql_adapter/foreigner_methods.rb:100:in `add_foreign_key'
/Users/bruceb/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/create_index_concurrently.rb:129:in `add_foreign_key'
/Users/bruceb/Code/OSS/git-repos/HornsAndHooves-pg_saurus/spec/dummy/db/migrate/20120207150844_add_foreign_keys.rb:4:in `change'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:606:in `exec_migration'
/Users/bruceb/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/migration/set_role_method.rb:71:in `exec_migration_with_role'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:590:in `block (2 levels) in migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:589:in `block in migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:588:in `migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:765:in `migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:995:in `block in execute_migration_in_transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:1041:in `block in ddl_transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `block in transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/connection_adapters/abstract/transaction.rb:188:in `within_new_transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/connection_adapters/abstract/database_statements.rb:213:in `transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/transactions.rb:220:in `transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:1041:in `ddl_transaction'
/Users/bruceb/Code/OSS/git-repos/HornsAndHooves-pg_saurus/lib/pg_saurus/create_index_concurrently.rb:205:in `ddl_transaction_with_postponed_queries'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:994:in `execute_migration_in_transaction'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:956:in `block in migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:952:in `each'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:952:in `migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:820:in `up'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/migration.rb:798:in `migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/tasks/database_tasks.rb:137:in `migrate'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/gems/activerecord-4.2.0/lib/active_record/railties/databases.rake:44:in `block (2 levels) in <top (required)>'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/bin/ruby_executable_hooks:15:in `eval'
/Users/bruceb/.rvm/gems/ruby-2.1.5@pg_saurus/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by baburdick** [comment]
Test migrations succeed and tests pass in master against Rails 4.0.13 and Rails 4.1.9 without these changes.

**Comment by MityaLiu** [comment]
Removed fk support and fixed few methods.
https://github.com/MityaLiu/pg_saurus/commit/e3cc2e9d9b32dd8f01124c3ffbf002f522f5910a

File: lib/pg_saurus/create_index_concurrently.rb (hunk 1 above)
Comment (skarm61) [inline lib/pg_saurus/create_index_concurrently.rb]: proper_table_name moved into ActiveRecord::Migration. Now you can use
```
table_name = proper_table_name(table_name)
```
http://stackoverflow.com/questions/26622488/activerecordmigrator-proper-table-name-is-deprecated

**Comment by baburdick** [comment]
@hubrix: It's long after your need. But we've built from @MityaLiu's branch and have a PR that will merge in the next few days as PgSaurus v3. If it's still of interest, take a look.

  HornsAndHooves/pg_saurus#16

No promises; but we expect that Rails 5 support in PgSaurus v4 will come much quicker than Rails 4.2 support did.

**Comment by baburdick** [comment]
Closing, as this PR is now moot with the merge of HornsAndHooves/pg_saurus#16 . Thanks for starting the ball rolling.

**Pull request closed without merging**